crates/aiken-project/src/lib.rs

```diff
@@ -565,7 +569,7 @@ where
                         .retain(|def| !matches!(def, Definition::Test { .. }))
                 });
 
-            self.type_check(parsed_modules, Tracing::silent(), true, true)?;
```

Comment:
Type-checking is now done in sequence for all pruned modules. So instead of doing it here with different options, we defer it to later (when we also do type-check our own project modules).